crates/ruff_notebook/src/cell.rs

```diff
@@ -164,9 +165,19 @@ impl Cell {
                         | "xdel"
                         | "xmode"
                 )
-            })
-        {
-            return true;
+            }) {
+                // The second token must _not_ be an operator, like `=` (to avoid false positives).
+                // The assignment operators can never follow a magic. Some binary operators _can_,
```

Comment:
I think the correct wording would be that "The assignment operators can never follow an automagic [..]" because `foo = !pwd` is valid and `foo` will be the output of `pwd` command.